design-documents/graph-ql/coverage/catalog/compare-list.md

```diff
@@ -1,19 +1,85 @@
 # Use cases
-* Buyer can add a product to the compare list.
-* If Buyer logging in or creating an account, an active compare list should be transferred to the revealed customer account.
-* Loginned buyer can access the latest compare list if the list has not been cleared with the last session. 
-* The buyer can request a compare list which has to return as a bare minimum price, link to a product page, name, and SKU.
 
+## Guest scenario 
 
-* A Compare list ID is a client-side generated ID.
-* A new compare list could be created by calling `addItemsToCompareList`.
-* If a shopper calls  `addItemsToCompareList` with the new id the previous cart will be retired and replaced with a new one.
+* A buyer can create a new compare list for the selected 
+products by calling mutation `addItemsToCompareList` with 
+the list of product ids and ID which will identify the list.
+Compare list ID is client generated identifier.
+```graphql
+{
+    mutation {
+        addItemsToCompareList(id: "aefe3cc0-89ec-4e92-8eb0-ae6ac545c5e8", items: ["100123", "234567", "874321"])
+    }
+}
+```
+* Registered customer can retrieve an active list ID, as well as item in the list, from the customer query.
+```graphql
+{
+    query {
+        customer {
+            compare_list {
+                list_id
+                items {
+                    sku
+                }
+            }
+        }
+    }
+}
+```
+* If the registered customer does not have an active list then null will be returned.
+This means the client has to generate and send a new ID if the compare list functionality requested.
 
+* If the buyer calls addItemsToCompareList with a new ID the previous list will be disabled. 
```

Comment:
It cannot be done for guests because there is not relation between the old and the new list.